e2e-tests/demand-backup-eks-credentials-irsa/run

```diff
@@ -0,0 +1,135 @@
+policy_arn="arn:aws:iam::119175775298:policy/operator-testing-access-s3"
+role_name="$cluster-psmdb-access-s3-bucket"
+
+# Create policy. Already done, we don't need to do it every time. But all steps should be illustrated in the test
+#aws iam create-policy --policy-name operator-testing-allow-access-s3 --policy-document file://conf/s3-bucket-policy.json
+
+desc "create role"
+jq --arg eks_cluster_oidc "$eks_cluster_oidc" \
+	'.Statement[0].Principal.Federated = "arn:aws:iam::119175775298:oidc-provider/\($eks_cluster_oidc)" |
+    .Statement[0].Condition.StringEquals[($eks_cluster_oidc + ":aud")] = "sts.amazonaws.com"' \
+	$test_dir/conf/template.json >$test_dir/conf/role-trust-policy.json
+
+role_arn=$(aws iam create-role \
+	--role-name "$role_name" \
+	--assume-role-policy-document file://$test_dir/conf/role-trust-policy.json \
+	--description "Allow access to s3 bucket" \
+	--query "Role.Arn" \
+	--output text)
+
+desc "connect role and policy"
+aws iam attach-role-policy --role-name "$role_name" --policy-arn $policy_arn
+
+create_infra "$namespace"
+
+desc "create secrets and start client"
+kubectl_bin apply \
+	-f "$conf_dir/secrets.yml" \
+	-f "$conf_dir/client.yml"
+
+desc "create first PSMDB cluster $cluster"
+apply_cluster $test_dir/conf/$cluster.yml
+
+desc 'check if all 3 Pods started'
+wait_for_running $cluster 3
+
+desc 'check if service and statefulset created with expected config'
+compare_kubectl statefulset/$cluster
+
+desc "update service accounts for operator and default (our cluster uses this one)"
+
+kubectl_bin annotate serviceaccount default \
+	eks.amazonaws.com/role-arn="$role_arn" \
+	--overwrite
+
+kubectl_bin annotate serviceaccount percona-server-mongodb-operator \
+	eks.amazonaws.com/role-arn="$role_arn" \
+	--overwrite
+
+desc "restart operator and cluster"
+operator_pod=$(get_operator_pod)
+kubectl_bin delete pod $operator_pod
+
+kubectl_bin delete pod "$cluster-0"
+kubectl_bin delete pod "$cluster-1"
+kubectl_bin delete pod "$cluster-2"
+
+wait_for_running $cluster 3
+
+kubectl exec $cluster-0 -c backup-agent -- sh -c 'if [ -z "$AWS_ROLE_ARN" ]; then echo "Variable AWS_ROLE_ARN not set" && exit 1; else echo "Variable AWS_ROLE_ARN is set"; fi'
+
+desc 'create user'
+run_mongo \
+	'db.createUser({user:"myApp",pwd:"myPass",roles:[{db:"myApp",role:"readWrite"}]})' \
+	"userAdmin:userAdmin123456@$cluster.$namespace"
+sleep 2
+
+desc 'write data, read from all'
+run_mongo \
+	'use myApp\n db.test.insert({ x: 100500 })' \
+	"myApp:myPass@$cluster.$namespace"
+
+minikube_sleep
+
+desc "compare mongo cmd"
+compare_mongo_cmd "find" "myApp:myPass@$cluster-0.$cluster.$namespace"
+compare_mongo_cmd "find" "myApp:myPass@$cluster-1.$cluster.$namespace"
+compare_mongo_cmd "find" "myApp:myPass@$cluster-2.$cluster.$namespace"
+
+#desc "wait backup agent"
+wait_backup_agent $cluster-0
+wait_backup_agent $cluster-1
+wait_backup_agent $cluster-2
+
+backup_name_aws="backup-aws-s3"
+
+desc 'run backups'
+run_backup aws-s3
```

Comment:
Check/consider checking scheduled backup but not only on demand and pitr.

Reply:
schedule backup works too. I don't think that we need to add it case to the test.